src/openrct2/drawing/X8DrawingEngine.cpp

```diff
@@ -403,24 +403,34 @@ void X8DrawingEngine::DrawAllDirtyBlocks()
             uint32_t columns = xx - x;
 
             // Check rows
-            uint32_t yy;
-            for (yy = y; yy < dirtyBlockRows; yy++)
+            uint32_t yy = y;
+            if (CheckRows(yy, y, xx, x, dirtyBlockRows, dirtyBlockColumns, columns, dirtyBlocks))
```

Comment:
After running my tests you do not need an if here. The next part of the code should always be run.